### PR TITLE
Fix train displaying as not honored despite respecting times & stop time

### LINKED ISSUE #8654: A train with respected times & stops is displayed "not honored"
### What happened?

The train 4 is displayed not honored 
image

### What did you expect to happen?

All scheduling constraints are respected so the train should be "Honored"

### How can we reproduce it (as minimally and precisely as possible)?

For Rec : Check this scneario : https://rec-osrd.reseau.sncf.fr/operational-studies/projects/6/studies/146/scenarios/276
For local : 
- Import [train_schedules.json](https://github.com/user-attachments/files/16818092/train_schedules.json)
- Launch this collection of RS : [Rolling stock.postman_collection.json](https://github.com/user-attachments/files/16818140/Rolling.stock.postman_collection.json)
- Verify Train 4

### On which environments the bug occurs?

Local, Recette (SNCF)

### On which browser the bug occurs?

Firefox, Google Chrome

### OSRD version (top right corner `Account` button > `Informations`)

80973c5

### PR DESCRIPTION
**Pull request by kmer2016**
closes #8654

### CODE REVIEW

File: front/src/applications/operationalStudies/utils.ts [hunk 1]
```diff
@@ -235,12 +235,17 @@ export const isScheduledPointsNotHonored = (
   }
 
   if (!trainSchedule.schedule) return false;
-  return trainSchedule.schedule.some((schedule, index) => {
+  return trainSchedule.schedule.some((schedule) => {
     if (!schedule.arrival) return false;
-    // need to add +1 because origin will never be in schedules
+    const matchindIndex = trainSchedule.path.findIndex((pathItem) => pathItem.id === schedule.at);
```
Comment (theocrsb) [inline front/src/applications/operationalStudies/utils.ts]: This solution gives quadratic complexity. I don't know if it's a problem, but when we set up this logic we thought about this complexity problem.
@flomonster, what do you think?

Comment (flomonster) [reply front/src/applications/operationalStudies/utils.ts]: Complexity can easily be improved. Just create a mapping from “pathItem.id” -> “pathItem”.

This way you can access the path item with its ID and check whether it was found.

**Comment by Maymanaf** [comment]
Works fine for me. Good job !

**Review by Maymanaf — APPROVED** [review]

**Review by flomonster — APPROVED** [review]

**Review by Caracol3 — APPROVED** [review]
Lgtm, good job 👌

**Review by theocrsb — APPROVED** [review]
LGTM

**Review by SharglutDev — APPROVED** [review]
Lgtm

**Pull request merged**